code/modules/food_and_drinks/food/snacks.dm

```diff
@@ -338,12 +342,28 @@ All foods are distributed among various categories. Use common sense.
 /// All the food items that can store an item inside itself, like bread or cake.
 /obj/item/reagent_containers/food/snacks/store
 	w_class = WEIGHT_CLASS_NORMAL
-	var/stored_item = 0
+	/// If an item has been stored in the food
+	var/stored_item = FALSE
+	/// The amount of volume the food has on creation
+	var/volume_on_creation = 0
+	/// Allows someone to bypass the small weight class requirement, so they can put whatever they want into a slice of bread
+	var/bypass_weight_limit = FALSE
+
+/obj/item/reagent_containers/food/snacks/store/Initialize()
+	. = ..()
+	if(reagents)
+		if(reagents.total_volume)
+			volume_on_creation = reagents.total_volume
 
 /obj/item/reagent_containers/food/snacks/store/attackby(obj/item/W, mob/user, params)
 	..()
-	if(W.w_class <= WEIGHT_CLASS_SMALL & !istype(W, /obj/item/reagent_containers/food/snacks)) //can't slip snacks inside, they're used for custom foods.
-		if(W.get_sharpness())
+	if(istype(W, /obj/item/reagent_containers/food/snacks)) //can't slip snacks inside, they're used for custom foods.
+		return 0
+
+	if((bypass_weight_limit || W.w_class <= WEIGHT_CLASS_SMALL)) 
```

Comment:
can you check out the rest of the bad usage of 0 and 1 instead of FALSE and TRUE in this proc, its a bit difficult to add suggestions for each